pyvelox/substrait.cpp

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "substrait.h" // @manual
+
+#include <google/protobuf/util/json_util.h>
+#include <velox/common/base/Exceptions.h>
+#include <velox/exec/tests/utils/AssertQueryBuilder.h>
+#include <velox/exec/tests/utils/HiveConnectorTestBase.h>
+#include <velox/exec/tests/utils/TempDirectoryPath.h>
+#include <velox/functions/prestosql/aggregates/RegisterAggregateFunctions.h>
+#include <velox/functions/prestosql/registration/RegistrationFunctions.h>
+
+#include <algorithm>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+
+#include "context.h"
+
+namespace facebook::velox::py {
+
+namespace py = pybind11;
+
+namespace {
+
+bool isFilePath(const std::string& path) {
+  return std::filesystem::exists(path);
+}
+
+static void readFromJSON(
```

Comment:
nit: would be better to name this something like validateJson as it looks like thats what we are doing.

Reply:
Sure, that's a good point.